src/mono/mono/mini/monovm.c

```diff
@@ -123,7 +123,7 @@ mono_core_preload_hook (MonoAssemblyLoadContext *alc, MonoAssemblyName *aname, c
 	size_t basename_len = strlen (basename);
 
 	for (int i = 0; i < a->assembly_count; ++i) {
-		if (basename_len == a->basename_lens [i] && !strncmp (basename, a->basenames [i], a->basename_lens [i])) {
```

Comment:
shouldn't it ignore case only on Windows? Or it doesn't make sense?

Reply:
Definitely something that could be done, but didn't want to start specialising it for Windows only, but if we would like to preserve case sensitive file name compare against the TPA list on other platforms, we can easily make the change Windows only.